presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/TestCachingHiveMetastore.java

```diff
@@ -171,22 +172,55 @@ public void testGetPartitionNamesByParts()
         assertEquals(mockClient.getAccessCount(), 2);
     }
 
+    private static class MockPartitionMutator
+            implements PartitionMutator
+    {
+        public enum VersionType
+        {
+            CONSTANT,
+            INCREASING,
+            DECREASING
+        }
+
+        private final VersionType versionType;
+        private int version = 1000;
+
+        public MockPartitionMutator(VersionType versionType)
```

Comment:
I'm thinking if we could just take a lambda here so that we don't need this enum:

`public MockPartitionMutator(Function<Integer, Integer> versionUpdater)` <- it is a poor name lol

So the caller looks like `new MockPartitionMutator(version -> version + 1)`

`mutate` would be something like:
```
version = versionUpdater.apply(version);
builder.setPartitionVersion(version);
```

Maybe it would make it easier? :)